packages/remix-dev/compiler/js/compiler.ts

```diff
@@ -94,15 +95,7 @@ const createEsbuildConfig = (
     emptyModulesPlugin(ctx, /^@remix-run\/(deno|cloudflare|node)(\/.*)?$/, {
       includeNodeModules: true,
     }),
-    nodeModulesPolyfillPlugin({
-      // For the browser build, we replace any Node built-ins that don't have a
-      // polyfill with an empty module. This ensures the build can pass without
-      // having to mark all Node built-ins as external which can result in other
-      // issues, e.g. https://github.com/remix-run/remix/issues/5521. We then
-      // rely on tree-shaking to remove all unused polyfills and fallbacks.
-      fallback: "empty",
-    }),
-    externalPlugin(/^node:.*/, { sideEffects: false }),
```

Comment:
Since [esbuild-plugins-node-modules-polyfill](https://github.com/imranbarbhuiya/esbuild-plugins-node-modules-polyfill) is now configured to manage fallbacks, this usage of `externalPlugin` was redundant.